fife/__main__.py

```diff
@@ -74,11 +82,33 @@ def main():
     print(f"Model training time: {time() - checkpoint_time} seconds")
 
     checkpoint_time = time()
-
     if test_intervals > 0:
 
         # Save metrics
-        max_test_intervals = (len(set(modeler.data["_period"])) - 1) / 2
+        max_test_intervals = int((len(set(modeler.data["_period"])) - 1) / 2)
+
+        if by_feature is not None:
+            values = list(set(data[by_feature]))
+
+            for feature_value in values:
+
+                evaluation_subset_by_period = modeler.data["_period"] == (
+                    modeler.data["_period"].max()
+                    - min(test_intervals, max_test_intervals)
+                )
```

Comment:
If we moved lines 100-102 above line 90, could we avoid subsetting with each iteration of the for loop?

Reply:
Yes! Good catch, I didn't even think about that! I made the change, am running build packages and then will commit it to the current branch